pkg/assembler/backends/inmem/artifact.go

```diff
@@ -43,14 +43,49 @@ type artStruct struct {
 
 func (n *artStruct) ID() uint32 { return n.id }
 
-func (n *artStruct) Neighbors() []uint32 {
-	out := make([]uint32, 0, len(n.hashEquals)+len(n.occurrences)+len(n.hasSLSAs)+len(n.vexLinks)+len(n.badLinks)+len(n.goodLinks))
-	out = append(out, n.hashEquals...)
-	out = append(out, n.occurrences...)
-	out = append(out, n.hasSLSAs...)
-	out = append(out, n.vexLinks...)
-	out = append(out, n.badLinks...)
-	out = append(out, n.goodLinks...)
+func (n *artStruct) Neighbors(allowedEdges edgeMap) []uint32 {
+	maxLen := 0
+	if allowedEdges[model.EdgeHashEqual] {
+		maxLen = maxLen + len(n.hashEquals)
+	}
+	if allowedEdges[model.EdgeIsOccurrence] {
+		maxLen = maxLen + len(n.occurrences)
+	}
+	if allowedEdges[model.EdgeHasSlsa] {
+		maxLen = maxLen + len(n.hasSLSAs)
+	}
+	if allowedEdges[model.EdgeCertifyVexStatement] {
+		maxLen = maxLen + len(n.vexLinks)
+	}
+	if allowedEdges[model.EdgeCertifyBad] {
+		maxLen = maxLen + len(n.badLinks)
+	}
+	if allowedEdges[model.EdgeCertifyGood] {
+		maxLen = maxLen + len(n.goodLinks)
+	}
+	if maxLen == 0 {
+		return []uint32{}
+	}
```

Comment:
Is the maintenance burden of the code to calculate the slice allocation size worth the perf improvement? It was neat when it was just 1 line.